src/main/java/com/vmware/vipclient/i18n/base/cache/MessageCache.java

```diff
@@ -43,13 +46,35 @@ public void setId(String id) {
     }
 
     @SuppressWarnings("unchecked")
-    public Map<String, String> get(String cacheKey) {
+    public CacheItem get(String cacheKey) {
         Integer i = hitMap.get(cacheKey);
         if (i != null) {
             hitMap.put(cacheKey, i.intValue() + 1);
         }
-        Object cachedObject = cachedComponentsMap.get(cacheKey);
-        return cachedObject == null ? null : (Map<String, String>) cachedObject;
+        return cachedComponentsMap.get(cacheKey);
+    }
+    
+    public boolean isExpired(String cacheKey) {
+    	CacheItem cacheItem = cachedComponentsMap.get(cacheKey);
```

Comment:
Do you think it's better to add this at the beginning of the method to make sure that the 2 mechanisms aren't conflicted?
if (VIPCfg.getInstance().getCacheExpiredTime() != VIPCfg.cacheExpiredTimeNotSet) {
    		return false;
}